### PR TITLE
Feat: add dependency cruiser rule in bare example

### PR DESCRIPTION
**Pull request by guillaumeduboc**

### CODE REVIEW

File: examples/swarmion-bare/commonConfiguration/dependency-cruiser.config.js [hunk 1]
```diff
@@ -12,6 +12,28 @@ module.exports = ({ pathNot, path } = { pathNot: [], path: [] }) => ({
...
+      name: 'no-application-in-provisioning',
+      severity: 'error',
+      comment:
+        'Please do not import application code in your provisioning code.',
+      from: { path: '^resources|^functions.+config' },
+      to: {
+        // you can remove contracts if you only use shared contracts
+        pathNot: '^resources|^functions.+config|^contracts',
+      },
+    },
```
Comment (fargito) [inline examples/swarmion-bare/commonConfiguration/dependency-cruiser.config.js:36]: You could add the same rules in the other starters, WDYT?

Comment (guillaumeduboc) [reply examples/swarmion-bare/commonConfiguration/dependency-cruiser.config.js:36]: sure
should I rename the `functions/index.ts` to `functions/config.ts` as well ?

Comment (fargito) [reply examples/swarmion-bare/commonConfiguration/dependency-cruiser.config.js:36]: I think it could better indeed

**Review by fargito — APPROVED** [review]
LGTM

**Pull request merged**